### ISSUE TITLE
Couldn't find an alternative telinit implementation to spawn

### ISSUE THREAD

**Issue by codetrainee** [rstudio/rstudio#4367]
Hi, 

I am running the Rstudio server on my windows subsystem linux (Ubuntu 18.04.1 LTS) and encountered a problem saying: 
```
Couldn't find an alternative telinit implementation to spawn
```
AFAIK, Ubuntu has switched to  `systemd` ecosystem and `telinit` in wsl (ubuntu 18.04) is a symbolic link pointing to ` /bin/systemctl*`, which could be the cause. 

Therefore, I was wondering if you have any plan to solve the compatibility.

**Comment by kevinushey** [comment]
We don't currently plan to support RStudio Server on WSL.

**Comment by jmcphers** [comment]
We'll use this request to track that.

**Comment by abalter** [comment]
More and more folks are using WSL, so this is important. Also, Microsoft has recently updated the way WSL installs, including a better filesystem. So looking into that would be useful.

https://www.windowscentral.com/whats-new-windows-console-windows-10-may-2019-update

https://devblogs.microsoft.com/commandline/announcing-wsl-2/

**Comment by black-snow** [comment]
I +1 this. I have all my tools in the WSL but I'd like to use RStudio. I don't want to keep another R installation in my windows machine + alle the packages just for that, though.

**Comment by tingjhenjiang** [comment]
the error "[rserver] ERROR system error 13 (Permission denied); OCCURRED AT: rstudio::core::Error rstudio::core::http::initTcpIpAcceptor(rstudio::core::http::SocketAcceptorService<rstudio_boost::asio::ip::tcp>&, const string&, const string&) /var/lib/jenkins/workspace/IDE/open-source-pipeline/v1.2-patch/src/cpp/core/include/core/http/TcpIpSocketUtils.hpp:103; LOGGED FROM: int main(int, char* const*) /var/lib/jenkins/workspace/IDE/open-source-pipeline/v1.2-patch/src/cpp/server/ServerMain.cpp:502" may be related to this issue?

**Comment by cupofjava2** [comment]
See this post and it works for me (though I found Xming is not required on Windows): https://johnmuschelli.com/neuroc/windows_wsl/index.html

**Comment by gtritchie** [comment]
> See this post and it works for me (though I found Xming is not required on Windows): https://johnmuschelli.com/neuroc/windows_wsl/index.html

RStudio Server is a pure web-based application, not sure why that article says to install X. That would be needed only if you wanted to run Linux Desktop version of RStudio inside WSL (which seems pointless).

**Comment by gtritchie** [comment]
As soon as WSL v2 goes into general release (i.e. doesn't require joining Windows Insider program) I'll test out running RStudio Server again; don't want to declare RStudio Server supported on WSL until then.

**Comment by abalter** [comment]
@gtritchie I was searching around, and I can't find more details on when v2 will be released. It was in preview build 18917. Will it be in 20H1 or 20H2?

**Comment by gtritchie** [comment]
Not official, but looks like 20H1: https://www.howtogeek.com/438830/whats-new-in-windows-10s-20h1-update-arriving-spring-2020/

**Comment by gtritchie** [comment]
Here's similar info on a Microsoft page: https://docs.microsoft.com/en-us/windows-insider/at-home/whats-new-wip-at-home-20h1

**Comment by abalter** [comment]
I hope they release it soon! I was trying windows insider for a while, but I need more stability.

**Comment by jflam** [comment]
FYI we shipped Windows 10 2004 with WSL 2 support this week. Would be lovely to have RStudio running on WSL 2!

**Comment by gtritchie** [comment]
Yup, it's on my list. I was able to run RStudio Server on WSL 1 with few problems (other than performance).

**Comment by gtritchie** [comment]
Did a trial run today; At first blush, RStudio Server runs great on WSL2.

I tried Ubuntu 18 build of RStudio Server 1.3.959 on Ubuntu 20 WSL2 instance following our standard install instructions.

https://rstudio.com/products/rstudio/download-server/debian-ubuntu/

Then you just hit localhost:8787 in browser.

Still get that previously mentioned message at end of the install: `Couldn't find an alternative telinit implementation to spawn` so need to look into that, but isn't prevent RStudio Server from starting up.

Just initial test, but looking good so far.

**Comment by gtritchie** [comment]
After restarting WSL, do have to manually restart RStudio Server via "sudo rstudio-server start" so that's probably related to that error message.

**Comment by psychelzh** [comment]
@gtritchie Thanks. You are right. After restarting WSL, it will always report an error if I just call `rstudio-server start`. I should first call `sudo rstudio-server start`, after that I call `rstudio-server start &` and `exit` in order that Rstudio Server will run at background. It is not so for Rstudio Server of versions older than 1.3.

**Comment by gtritchie** [comment]
Support article published on running RStudio Server in WSL2. Problem with having to manually restart still there, but keeping this issue open to track investigation of that.

https://support.rstudio.com/hc/en-us/articles/360049776974-Using-RStudio-Server-on-Windows-with-WSL2

**Comment by nickschurch** [comment]
Just to add, RStudio works on WSL if you install via conda - but the conda package is out of date and is officially unsupported so Your Milage May Vary.

**Comment by aborruso** [comment]
Hi all,
when I run `sudo rstudio-server start` I have 

```
TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/etc/rstudio/logging.conf'. Logging to '/var/log/rstudio/rstudio-server/rserver.log'.
```

How to solve it?

A note: I do not have these 2 files on my debian WSL2 OS

**Comment by aborruso** [comment]
> Hi all, when I run `sudo rstudio-server start` I have
> 
> ```
> TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/etc/rstudio/logging.conf'. Logging to '/var/log/rstudio/rstudio-server/rserver.log'.
> ```
> 
> How to solve it?
> 
> A note: I do not have these 2 files on my debian WSL2 OS

If I run http://localhost:8787/ it works. #solved

**Comment by abalter** [comment]
Please post your steps for installing rstudio-server. I've had no trouble. Also, is there a reason you are using rstudio-server instead of rstudio-desktop?
